blocks/graph.py

```diff
@@ -204,10 +204,15 @@ def replace(self, replacements):
         # Sort `replacements` in topological order
         for node in apply_nodes_sorted:
             for input_ in node.inputs:
-                if input_ not in replacements:
-                    continue
-                replacement_keys_cur.append(input_)
-                replacement_vals_cur.append(replacements[input_])
+                if input_ in replacements:
```

Comment:
It seems to me that the whole iteration could be actually replaced by `for variable in self.variables`. They are also sorted topologically, exactly in the way as done here (correct me if I am wrong), but the duplicates are removed, which is just what we need here.